src/core/sagas/products.ts

```diff
@@ -9,26 +11,71 @@ import * as utils from '../utils';
 import { Tasks as productDetailsTasks } from './product-details';
 
 export namespace Tasks {
-  export function* fetchProducts(flux: FluxCapacitor, action: Actions.FetchProducts) {
+  export function* fetchProductsAndNavigations(flux: FluxCapacitor, action: Actions.FetchProducts) {
     try {
-      const request = yield effects.select(Requests.search, flux.config);
-      const res = yield effects.call([flux.clients.bridge, flux.clients.bridge.search], request);
-
-      if (res.redirect) {
-        yield effects.put(flux.actions.receiveRedirect(res.redirect));
+      const [products, navigations]: [Results, Store.Recommendations.Navigation[]] = yield effects.all([
+        effects.call(fetchProducts, flux, action),
+        effects.call(fetchNavigations, flux, action)
+      ]);
+      if (products.redirect) {
+        yield effects.put(flux.actions.receiveRedirect(products.redirect));
       }
-
-      if (flux.config.search.redirectSingleResult && res.totalRecordCount === 1) {
-        yield effects.call(productDetailsTasks.receiveDetailsProduct, flux, res.records[0]);
+      if (flux.config.search.redirectSingleResult && products.totalRecordCount === 1) {
+        yield effects.call(productDetailsTasks.receiveDetailsProduct, flux, products.records[0]);
       } else {
-        flux.emit(Events.BEACON_SEARCH, res.id);
-        yield effects.put(<any>flux.actions.receiveProducts(res));
+        flux.emit(Events.BEACON_SEARCH, (<any>products).id);
+        if (navigations && !(navigations instanceof Error)) {
+          const iNav = flux.config.recommendations.iNav;
+          if (iNav.navigations.sort) {
+            products.availableNavigation =
+              RecommendationsAdapter.sortNavigations(products.availableNavigation, navigations);
+          }
+          if (Array.isArray(iNav.navigations.pinned)) {
+            // tslint:disable-next-line max-line-length
+            products.availableNavigation = RecommendationsAdapter.pinNavigations(products.availableNavigation, flux.config);
+          }
+          if (iNav.refinements.sort) {
+            products.availableNavigation =
+              RecommendationsAdapter.sortRefinements(products.availableNavigation, navigations);
+          }
+          if (iNav.refinements.pinned) {
+            // tslint:disable-next-line max-line-length
+            products.availableNavigation = RecommendationsAdapter.pinRefinements(products.availableNavigation, flux.config);
+          }
+          yield effects.put(<any>flux.actions.receiveRecommendationsNavigations(navigations));
```

Comment:
obviously keep the yield, but object transformation should all happen in external functions